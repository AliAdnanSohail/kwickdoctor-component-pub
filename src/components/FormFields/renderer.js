import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
