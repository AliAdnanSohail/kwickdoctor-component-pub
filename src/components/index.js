export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
