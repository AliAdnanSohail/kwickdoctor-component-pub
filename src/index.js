export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
