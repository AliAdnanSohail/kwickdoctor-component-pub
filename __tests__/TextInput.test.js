import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { TextInput } from '../src';

const baseProps = {
  id: 'text-input-id',
  name: 'text-input-name',
  onChange() {},
  placeholder: 'text-input-placeholder',
  value: 'text-input-value',
  label: 'text-input-label',
  error: '',
  className: 'text-input-class',
  multiline: false,
};

it('TextInput renders correctly', () => {
  const element = renderer.create(<TextInput {...baseProps} />).toJSON();

  expect(element).toMatchSnapshot();
});

it('TextInput set id correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find(`[htmlFor="${baseProps.id}"]`)).toHaveLength(1);
  expect(element.find(`input#${baseProps.id}`)).toHaveLength(1);
});

it('TextInput set name correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find(`input[name="${baseProps.name}"]`)).toHaveLength(1);
});

it('TextInput set placeholder correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find(`input[placeholder="${baseProps.placeholder}"]`)).toHaveLength(1);
});

it('TextInput set value correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find('input').instance().value).toEqual(baseProps.value);
});

it('TextInput set label correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find('label').text()).toEqual(baseProps.label);
});

it('TextInput set class correctly', () => {
  const element = mount(<TextInput {...baseProps} />);

  expect(element.find('input').hasClass(baseProps.className)).toEqual(true);
});

it('TextInput textarea renders correctly', () => {
  const props = Object.assign({}, baseProps);
  props.multiline = true;
  const element = mount(<TextInput {...props} />);

  expect(element.find('textarea')).toHaveLength(1);
});

it('TextInput input correctly', () => {
  const props = Object.assign({}, baseProps);
  const inputValue = 'Changed';
  let value = '';
  props.onChange = (event) => {
    value = event.target.value;
  };

  const element = mount(<TextInput {...props} />);
  const input = element.find('input');
  input.simulate('focus');
  input.simulate('change', { target: { value: inputValue } });

  expect(inputValue === value).toEqual(true);
});

it('TextInput show error correctly', () => {
  const props = Object.assign(baseProps);
  props.error = 'error-text';
  const element = mount(<TextInput {...props} />);

  expect(element.find('.input-error').text()).toEqual(props.error);
});
