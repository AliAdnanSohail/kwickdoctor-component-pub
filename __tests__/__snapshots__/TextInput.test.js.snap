// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<form>
  <div
    className="jsx-2941157369 jsx-1178708339"
  >
    <label
      className="jsx-2941157369 jsx-1178708339"
      htmlFor="text-input-id"
    >
      text-input-label
    </label>
    <input
      className="jsx-2542774814 text-input-class"
      id="text-input-id"
      name="text-input-name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="text-input-placeholder"
      type="text"
      value="text-input-value"
    />
    <style>
      input,textarea{width:100%;min-width:100%;max-width:100%;min-height:44px;padding:12px 16px;background-color:#ffffff;color:#2A2A30;box-sizing:border-box;font-style:normal;font-weight:normal;font-size:0.875rem;border-width:2px;border-style:solid;border-color:#E7E7F0;border-radius:4px;}
      input::-webkit-input-placeholder,textarea::-webkit-input-placeholder{color:#828393;}
      input::-moz-placeholder,textarea::-moz-placeholder{color:#828393;}
      input:-ms-input-placeholder,textarea:-ms-input-placeholder{color:#828393;}
      input:-moz-placeholder,textarea:-moz-placeholder{color:#828393;}
      input:focus,textarea:focus{outline:none;padding:12px 16px;border-color:#0c97f9;}
      input.error,input.error:focus,textarea.error,textarea.error:focus{border-color:#FF5572;}
    </style>
  </div>
</form>
`;
